pubspec.yaml

```diff
@@ -24,6 +24,7 @@ dependencies:
   shelf: ^1.1.1
   source_span: ^1.8.1
   stack_trace: ^1.10.0
+  stream_transform: ^2.0.0
```

Comment:
I don't see it here: https://github.com/dart-lang/sdk/blob/main/DEPS

So we would need to add a dependency on it, the question becomes if it isn't less work to do:
```dart
transformStream(Stream<List<int>> stream) async* {
  await for (final chunk in stream) {
    crc.update(chunk);
    yield chunk;
  }
}
```

There might be dragons here I haven't thought of, but I suspect it's not super dangerous to do this.

Reply:
Oh yea, that looks like it should be a good alternative. I'll work on implementing that.

Reply:
I ended up using a [conceptually similar approach](https://github.com/dart-lang/pub/pull/3546/commits/7755d07bbf4c736aa12567c298e2fb8f6d42149c) which should be less dragon-y. 😃